dataflow/gpu-examples/conftest.py

```diff
@@ -0,0 +1,508 @@
+                        "gcloud",
+                        "builds",
+                        "submit",
+                        f"--project={project}",
+                        f"--config={config}",
+                        *cmd_substitutions,
+                        source,
+                    ]
+                    logging.info(f"{cmd}")
+                    subprocess.run(cmd, check=True)
+                    logging.info(f"Cloud build finished successfully: {config}")
+                    yield f.read()
+            except Exception as e:
+                logging.exception(e)
+                logging.warning(f'Current directory: {os.getcwd()}')
+                yield config
+        elif image_name:
+            cmd = [
+                "gcloud",
+                "builds",
+                "submit",
+                f"--project={project}",
+                f"--tag=gcr.io/{project}/{image_name}:{UUID}",
+                *cmd_substitutions,
+                source,
+            ]
+            logging.info(f"{cmd}")
+            subprocess.run(cmd, check=True)
+            logging.info(f"Created image: gcr.io/{project}/{image_name}:{UUID}")
+            yield f"{image_name}:{UUID}"
+        else:
+            raise ValueError("must specify either `config` or `image_name`")
+
+        if image_name:
+            cmd = [
+                "gcloud",
+                "container",
+                "images",
+                "delete",
+                f"gcr.io/{project}/{image_name}:{UUID}",
+                f"--project={project}",
+                "--force-delete-tags",
+                "--quiet",
+            ]
+            logging.info(f"{cmd}")
+            subprocess.run(cmd, check=True)
+            logging.info(f"Deleted image: gcr.io/{project}/{image_name}:{UUID}")
+
+    @staticmethod
+    def dataflow_jobs_list(
+        project: str = PROJECT, page_size: int = 30
+    ) -> Iterable[dict]:
+        from googleapiclient.discovery import build
+
+        dataflow = build("dataflow", "v1b3")
+
+        response = {"nextPageToken": None}
+        while "nextPageToken" in response:
+            # For more info see:
+            #   https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs/list
+            request = (
+                dataflow.projects()
+                .jobs()
+                .list(
+                    projectId=project,
+                    pageToken=response["nextPageToken"],
+                    pageSize=page_size,
+                )
+            )
+            response = request.execute()
+            for job in response["jobs"]:
+                yield job
+
+    @staticmethod
+    def dataflow_jobs_get(
+        job_id: Optional[str] = None,
+        job_name: Optional[str] = None,
+        project: str = PROJECT,
+        list_page_size: int = 30,
+    ) -> Optional[Dict[str, Any]]:
+        from googleapiclient.discovery import build
+
+        dataflow = build("dataflow", "v1b3")
```

Comment:
If we update past v1b3 will this build command fail? (just making sure it's future proofed so that when dependencies get updated we don't break samples)

Reply:
I don't think it would break, but I don't think it receives a ton of maintenance either. I personally prefer the gcloud commands for these things, but I know the general rule is to avoid them if there's a client library.